ios/Classes/Plugin/PluginController.swift

```diff
@@ -34,13 +35,14 @@ final class PluginController {
             onDiscovery: papply(weak: self) { context, _, peripheral, advertisementData, rssi in
                 guard let sink = context.scan?.sink
                 else { assert(false); return }
-                
+
                 let serviceData = advertisementData[CBAdvertisementDataServiceDataKey] as? ServiceData ?? [:]
                 let manufacturerData = advertisementData[CBAdvertisementDataManufacturerDataKey] as? Data ?? Data();
+                let localName = advertisementData[CBAdvertisementDataLocalNameKey] as? String ?? String();
```

Comment:
The documentation on [CBPeripheral.name](https://developer.apple.com/documentation/corebluetooth/cbperipheral/1519029-name) states:

> A peripheral may have two different name types: one that the device advertises and another that the device publishes in its database as its Bluetooth low energy Generic Access Profile (GAP) device name. If a peripheral has both types of names, this property returns its GAP device name.

Which suggests a name may be missing in the advertisement (e.g. depending on the device mode). I would suggest to fall back to `.name` in such a case:

```swift
let name = advertisementData[CBAdvertisementDataLocalNameKey] as? String ?? peripheral.name ?? String();
```

This way the library reverses the prioritization of `CBPeripheral.name` and may show a device name different from what a user can see in the system settings, which I don't find as an unconditionally useful change, but I agree that the aggressive caching and/or insufficient cache invalidation in the iOS BLE stack does seem to be more harmful from the hardware and/or software developer standpoint.